### ISSUE TITLE
Router gets resolved for empty url.

### ISSUE THREAD

**Issue by loopingLines** [krasimir/navigo#31]
Hi,

First, Thankyou for the simple router. Made my life easy.

I am not reporting a bug but need some help here. This should have been very simple but I am not able to figure out how to differentiate between no query and an unmatched query. I have written a fallback so:
router.on({
	'p1' : loadP1,
	'p2' : loadP2,
	'g1' : loadG1,
	's1' : loadS1,
	'user' : loadUserPage,
        '#': loadLanding
}).resolve();
router.on(function fallback() {
	alert('oops!! this page was not found');
}).resolve();
but this get fired even for my landing index.html without any queries at all. I want to fire the fallback for unmatched queries only. I am using hash.

**Comment by krasimir** [comment]
Hey,

Navigo falls back to it's default handler if the page does not exists. To achieve what you want you have to define a sub path. For example:

```js
router.on({
  'website/p1': function (params) {
    console.log('p1');
  },
  'website/p2': function (params) {
    console.log('p2');
  },
  'website/*': function (params) {
    console.log('not found');
  }
});
router.on(function defaultRoute() {
  console.log('default');
});
```

I agree that maybe the router should handle those cases in a different way. I.e. if there is no url path then falls back to its default handler but if there is a path and if it doesn't match then fires another callback.

**Comment by loopingLines** [comment]
Unfortunately, this did not work for me.  Also, in my case the origin as well as pathname could change. However, before setting up the router if I wrote this, the router works as expected.

```javascript
if(window.location.href.split('#')[1] == undefined) {
	window.location.hash = '#';
}
```
There is something else I found out. This is how I set up the router:
```javascript
router.on({
	'page1' : loadPage1,
	'page2' : loadPage2,
	'user' : loadUserPage,
	'#' : loadRoot
}).resolve();
router.on(function fallback() {
	alert('oops!! this page was not found');
}).resolve();
```
Story1:
If I write the full location of my page in browser's location bar like this "../main/index.html", with the above script in place, the location is changed to "../main/index.html#" and loadRoot is fired. Win.

But if the location is so  "../main/", it changes to "../main/#" and the fallback is fired.

Story2:
If the page lands on any one of the urls directly, Example:- "../main/index.html#page1" , No matter how I navigate to "../main/index.html#", loadRoot is not fired but fallback is.
Somehow, the roots are different when there is a query url and when there is not. There is an extra hash in the root in the former case. I solved this one by passing the root(without hash)  while instantiating Navigo.

I guess may be the no url case will have to be handled after all.

**Comment by krasimir** [comment]
Yep, definitely we need different handlers for `no-match` and `root-url` cases. I'll leave this open and I hope will have time to implement these changes soon.

**Comment by krasimir** [comment]
Just released a new `2.2.0` version. We now have a [`notFound`](https://github.com/krasimir/navigo#not-found-handler) method that handles cases where a path is specified but not matching any of the registered routes. The current default/fallback route is called only if someone loads the page with no path specified and there is no matching rule for that. The README is updated.
